### PR TITLE
unpin numpy

### PR DESCRIPTION
**Pull request by maxime-desroches**

### CODE REVIEW

File: .github/workflows/selfdrive_tests.yaml [hunk 1]
```diff
@@ -117,6 +117,8 @@ jobs:
         restore-keys: |
           scons-${{ runner.arch }}-macos-${{ env.CACHE_COMMIT_DATE }}
           scons-${{ runner.arch }}-macos
+      # make new cache
+    - run: rm -rf /tmp/scons_cache/*
```
Comment (adeebshihadeh) [inline .github/workflows/selfdrive_tests.yaml:121]: we need to fix this scons dependency properly

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

File: .github/workflows/selfdrive_tests.yaml (hunk 1 above)
Comment (adeebshihadeh) [reply .github/workflows/selfdrive_tests.yaml:121]: good example of why we need to do this properly, from latest master: https://github.com/commaai/openpilot/actions/runs/10151615366/job/28071268349

File: site_scons/site_tools/cython.py [hunk 2]
```diff
@@ -51,6 +52,7 @@ def create_builder(env):
   return cython
 
 def cython_suffix_emitter(env, source):
+  env.Depends(str(source[0]).split('.')[0] + env["CYTHONCFILESUFFIX"], np.__file__)
```
Comment (adeebshihadeh) [inline site_scons/site_tools/cython.py:55]: you sure you don't want to add this to the scanner?

**Comment by maxime-desroches** [comment]
Most packages still don't support numpy >= 2.0.0 (rerun for example)

**Pull request merged**